iree/compiler/Conversion/LinalgToLLVM/LinalgTileAndVectorizePass.cpp

```diff
@@ -61,11 +68,39 @@ struct TileAndVectorizeWorkgroups
 };
 }  // namespace
 
+namespace {
+/// Pattern to promote all matmul operands to memory.
+struct PromoteMatmulSubviewsPattern
+    : public linalg::LinalgPromotionPattern<linalg::MatmulOp> {
+  PromoteMatmulSubviewsPattern(MLIRContext *context,
+                               linalg::LinalgPromotionOptions options,
+                               linalg::LinalgTransformationFilter marker,
+                               PatternBenefit benefit = 1)
+      : linalg::LinalgPromotionPattern<linalg::MatmulOp>(
+            context,
+            options.setOperandsToPromote({0, 1, 2})
+                .setUseFullTileBuffersByDefault(true),
+            marker, benefit) {}
+};
+}  // namespace
+
 void TileAndVectorizeWorkgroups::runOnFunction() {
   auto funcOp = getOperation();
   MLIRContext *context = &getContext();
   CPUKernelDispatch cpuKernelDispatch;
 
+  // Promotes workgroups subviews to a full-tile allocated on the stack.
+  if (clEnablePromoteWorkgroupToFullTiles) {
+    OwningRewritePatternList promotionPatterns;
+    promotionPatterns.insert<PromoteMatmulSubviewsPattern>(
```

Comment:
Instead of doing this, its better to use a custom allocation function. Something similar to what is done on the GPU side....

Reply:
Done! will move the functions into a util in follow up PR.

Reply:
Thanks!